src/OrchardCore/OrchardCore.DisplayManagement/Razor/RazorViewCompiledItem.cs

```diff
@@ -1,24 +1,25 @@
 using System;
 using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc.Razor.Extensions;
 using Microsoft.AspNetCore.Razor.Hosting;
 
 namespace OrchardCore.DisplayManagement.Razor
 {
-    public class RazorViewCompiledItem : RazorCompiledItem
+    public sealed class RazorViewCompiledItem : RazorCompiledItem
     {
-        public RazorViewCompiledItem(Type type, string kind, string identifier, object[] metadata = null)
+        private object[] _metadata;
+
+        public RazorViewCompiledItem(Type type, string identifier, object[] metadata = null)
         {
             Type = type;
-            Kind = kind;
             Identifier = identifier;
-            Metadata = metadata ?? Array.Empty<object>();
```

Comment:
@jtkech You're losing the metadata here now

Reply:
It's a copy of the aspnetcore internal `DefaultRazorViewCompiledItem` whose ctor doesn't have a metadata parameter, the Metadata property being auto built if null from assembly attributes.